### PR TITLE
Fixed months to be in english. Added make_index journal function

### PR DESCRIPTION
**Pull request by kwrobert**
I kind of missed the `:VimwikiDiaryGenerateLinks` function from Vimwiki, so I added something similar here. First shot at writing any vimscript, so if everything looks good I'll add a proper command for people to use.

I did have a question, which is about how one should deal with joining paths in vimscript in a platform-agnostic way. Is there a community-standard way of joining strings into a path that is platform-independent? Something like Python's `os.path.join`?

### CODE REVIEW

**Comment by lervag** [comment]
Hi @kwrobert! Thank you for the PR!

First, I think you are addressing two issues with this. I would prefer if you could separate this, because I don't think your solution regarding month names is correct, although I agree it is an issue. If you open a new issue with a very short description of this, then I can implement a better solution that allows others (e.g. me) to keep a localised version of month names.

Regarding your question, I don't really think there is anything like python path joining in Vim. But I also think that things typically work as expected if you just always use forward slashes. I.e., you can write expressions like  `l:root . '/' . l:filename` or similar, and it should still work on Windows in most cases. IIRC.

I've also added a couple of comments to your code.

File: autoload/wiki/journal.vim [hunk 1]
```diff
@@ -93,3 +129,5 @@ function! s:get_links_generic(rx, fmt) abort " {{{1
 endfunction
 
 " }}}1
+"
```
Comment (lervag) [inline autoload/wiki/journal.vim]: Please remove extra lines.

File: autoload/wiki/journal.vim [hunk 2]
```diff
@@ -54,7 +54,43 @@ function! wiki#journal#go_to_month() abort " {{{1
...
+      put =''
+      for entry in entries
+        put ='- ['.entry.']('.entry.')'
+      endfor
+      put =''
+    endfor
+  endfor
+endfunction
+
+" }}}1
```
Comment (lervag) [inline autoload/wiki/journal.vim]: Please separate the public functions from the private functions with a single line here.

File: autoload/wiki/journal.vim [hunk 3]
```diff
@@ -54,7 +54,43 @@ function! wiki#journal#go_to_month() abort " {{{1
 endfunction
 
 " }}}1
+function! wiki#journal#make_index() " {{{1
+  let l:regex_days = '\d\{4}-\d\d-\d\d'
+  let entries=s:get_links_generic(l:regex_days, '%Y-%m-%d')
```
[the remark this answers is not in the document]
Comment (lervag) [reply autoload/wiki/journal.vim]: Note, both comments need more changes in later code as well.

**Comment by lervag** [comment]
Please see the comments. I should probably have used the review feature, but I think you find the comments if you click "File changes".

Note, I think the code generally looks very good, and the comments are a little bit pedantic. Still, I very much like to keep my code consistent, and I hope you can update as asked, then rebase and force push.

Let me know if you need any further assistance on this.

**Comment by kwrobert** [comment]
No problem, consistency is important! 

I've added the requested modifications, quick question though, as I'm a bit new to contributing to open source. When you say "rebase and force push", do you mean squash these last two commits into a single commit then force push up to my remote repo to rewrite the history there?

And I agree my "solution" to the whole month issue was more just a reflection of my personal preference. I rolled that back to the version from your repo, I'll open up a separate issue so we can discuss a more general way to handle this.

File: autoload/wiki/journal.vim [hunk 4]
```diff
@@ -53,6 +53,43 @@ function! wiki#journal#go_to_month() abort " {{{1
...
+        let year_dict[month] = [entry]
+      endif
+    else
+      let sorted_entries[year] = {month:[entry]}
+    endif
+  endfor
+
+  for year in reverse(sort(keys(sorted_entries)))
+    let l:month_dict = sorted_entries[year]
+    put ='# '.year
```
Comment (lervag) [inline autoload/wiki/journal.vim]: Add spaces before/after the `.` (makes it easier to distinguish from dict members).

File: autoload/wiki/journal.vim [hunk 5]
```diff
@@ -53,6 +53,43 @@ function! wiki#journal#go_to_month() abort " {{{1
...
+    put =''
+    for [month, entries] in items(month_dict)
+      let l:mname = wiki#date#get_month_name(month)
+      let l:mname = toupper(mname[0]).mname[1:strlen(mname)]
```
Comment (lervag) [inline autoload/wiki/journal.vim]: Same as above.

File: autoload/wiki/journal.vim [hunk 6]
```diff
@@ -53,6 +53,43 @@ function! wiki#journal#go_to_month() abort " {{{1
...
+      put ='## '.mname
```
Comment (lervag) [inline autoload/wiki/journal.vim]: Ditto.

**Comment by lervag** [comment]
Yes, when I say "rebase and force push", I mean that I prefer to get logical changes in separate commits, and that such changes typically require that you do `git push -f` to force push the changes on your remote repo.

And thanks, I think this looks mostly good. I had a couple of minor comments that escaped me the first time.

**Comment by kwrobert** [comment]
Alright, I squashed all my new commits together. This should contain everything, for some reason Github closed my pull request after I force pushed. Mind reopening?

Also, two questions:

1) Should I add a command to expose this function to the user?
2) Right now the inserted journal links are relative, meaning they don't work if you were to insert them on any wiki page that isn't in the journal subdirectory. Should I fix this to make the links absolute?

**Comment by lervag** [comment]
Hmm, that's strange. I can't reopen this PR. Perhaps you could just open a new one and link to this for the description?

**Pull request closed without merging**